### PR TITLE
Core/MotionMaster: cleanup, reordering and renaming

### PR DESCRIPTION
**Pull request by ccrs**
**Changes proposed:**
Some good renaming, reordering and cleaning.
No logic change at all.

**Target branch(es):** 3.3.5

### CODE REVIEW

**Comment by ccrs** [comment]
pretty pls, review fast so I can keep working on more stuff :dancer:

**Comment by funjoker** [comment]
Will be work against riztazz.

**Comment by ccrs** [comment]
dw, its separate things and he hardly touches MotionMaster

File: src/server/game/Movement/MotionMaster.h [hunk 1]
```diff
@@ -212,27 +165,26 @@ class TC_GAME_API MotionMaster //: private std::stack<MovementGenerator *>
...
-        uint8 _cleanFlag;
+        void DirectDelete(MovementGenerator* curr);
+        void DelayedDelete(MovementGenerator* curr);
+
+        ExpireList* m_expireList;
+        MovementGenerator* m_slot[MAX_MOTION_SLOT];
+        int m_top;
+        Unit* m_owner;
+        bool m_initialize[MAX_MOTION_SLOT];
+        uint8 m_cleanFlag;
```
Comment (joschiwald) [inline src/server/game/Movement/MotionMaster.h]: you are going backsteps

Comment (ccrs) [inline src/server/game/Movement/MotionMaster.h]: well there are 4 ways of naming used in the core
* m_myVariable
* myVariable
* _myVariable
* MyVariable

I just used the most common one in core classes

Comment (joschiwald) [inline src/server/game/Movement/MotionMaster.h]: _private
Public
Protected
local

**Comment by ccrs** [comment]
anything else?

File: src/server/game/Movement/MotionMaster.cpp [hunk 2]
```diff
@@ -122,84 +118,84 @@ void MotionMaster::UpdateMotion(uint32 diff)
...
 
-void MotionMaster::DelayedExpire()
+MovementGenerator* MotionMaster::GetMotionSlot(int slot) const
 {
-    if (size() > 1)
+    ASSERT(slot >= 0);
+    return _slot[slot];
+}
+
+void MotionMaster::propagateSpeedChange()
```
Comment (Temppus) [inline src/server/game/Movement/MotionMaster.cpp]: Maybe propagateSpeedChange -> PropagateSpeedChange ?

File: src/server/game/Movement/MotionMaster.cpp [hunk 3]
```diff
@@ -33,9 +32,18 @@
 #include "MoveSpline.h"
 #include "MoveSplineInit.h"
 
-inline bool isStatic(MovementGenerator *mv)
+inline bool IsStatic(MovementGenerator *mv) { return (mv == &si_idleMovement); }
```
Comment (tkrokli) [inline src/server/game/Movement/MotionMaster.cpp]: Please keep the original codestyle to highlight the actual change:
```cpp
inline bool IsStatic(MovementGenerator *mv)
{
	return (mv == &si_idleMovement);
}
```

Comment (ccrs) [inline src/server/game/Movement/MotionMaster.cpp]: https://github.com/TrinityCore/TrinityCore/blob/0c113b8731dc397f4ad30d375fd8be5f180d5728/src/server/game/Support/SupportMgr.cpp#L22 but whatever

Comment (ccrs) [inline src/server/game/Movement/MotionMaster.cpp]: and cant be delcared in the header, it'd need to add an include or a forward declaration of the movement generator

**Comment by ccrs** [comment]
so, somethin else? this is kinda a bottleneck :D

**Comment by Shauren** [comment]
Why? You should know already that codestyle-only changes are frowned upon.

File: src/server/game/Movement/MotionMaster.h [hunk 4]
```diff
@@ -80,91 +81,41 @@ enum RotateDirection
...
-        {
-            if (empty())
-                return;
+        typedef std::vector<MovementGenerator*> ExpireList;
 
-            Impl[_top] = NULL;
-            while (!empty() && !top())
-                --_top;
-        }
-        void push(_Ty _Val) { ++_top; Impl[_top] = _Val; }
```
Comment (jackpoz) [inline src/server/game/Movement/MotionMaster.h]: where did this function go ?

Comment (ccrs) [inline src/server/game/Movement/MotionMaster.h]: cleaning -> unused function -> erased

**Comment by ccrs** [comment]
so, anything else?

**Comment by bieleib92** [comment]
@ccrs for president :D

**Pull request merged**